include/flatbuffers/flatbuffers.h

```diff
@@ -1750,15 +1756,36 @@ class FlatBufferBuilder {
   /// where the vector is stored.
   template<typename T> Offset<Vector<T>> CreateVector(size_t vector_size,
       const std::function<T (size_t i)> &f) {
-    std::vector<T> elems(vector_size);
-    for (size_t i = 0; i < vector_size; i++) elems[i] = f(i);
-    return CreateVector(elems);
+    StartVector(vector_size, sizeof(T));
+    for (auto i = vector_size; i > 0;) { 
+      PushElement(f(--i)); 
```

Comment:
It looks strange. Why `--i` is not part of the `for` expression?

Reply:
Sounds good.